---
output:
  html_document:
    keep_md: no
  pdf_document: default
params:
  directory:
    value: x
  input_data:
    value: x
  stage:
    value: x
---
## Cluster-update workflow report

Brief description of the methods used in each step of the workflow, followed by main results.


```{r,results="hide",echo=FALSE,message=FALSE,warning=FALSE}
library(tidyverse)
library(data.table)
library(knitr)
library(kableExtra)
library(markdown)
library(rmarkdown)
```

### Data to be integrated:

Brief description of the initial data to be integrated in the cluster DB:

```{r,echo=FALSE,results='asis'}
dir <- params$directory
input_data <- params$input_data
stage <- params$stage
stagec <- params$stage == "contigs"
seqs <- system(paste0("cat ", input_data, " | grep -c '^>'"),intern = T)

cat(paste("The data in input are at the stage of ",stage,", and the number of sequences is: ", seqs))
```

```{r conditional_block, eval=stagec, echo=FALSE}
print("Since the data to be integrated in input are at the contig stage, we performed a gene prediction with Prodigal (Prokaryotic Dynamic Programming Gene-finding Algorithm) (version 2.6.3: February 2016) program [[1]](#1), applied in metagenomic mode.
The total number of predicted genes can be found in the following table.")

genes <- system(paste0("cat ",dir,"/gene_prediction/orf_seqs.txt", " | wc -l "),intern = T)
tibble(Contigs=contig_seqs,Genes=genes) %>% kable(digits=3, format.args = list(big.mark = ",")) %>%
  kable_styling(bootstrap_options = c("striped", "condensed"),full_width = F)

```

### Pfam annotations

We functionally annotated the genes using the Pfam database of protein domain families (version 30.0) [[2]](#2), with the hmmsearch program from the HMMER package (version: 3.1b2) [[3]](#3). We only accepted those hits with an e-value < 1e-5 and a coverage > 0.4.

```{r,echo=FALSE, warning=FALSE}
genes <- fread(paste0(dir,"/gene_prediction/orf_seqs.txt"), stringsAsFactors = F,header = F)
pfam_annot <- fread(paste0(dir,"/annot_and_clust/pfam_name_acc_clan_multi.tsv"),
stringsAsFactors = F,header = F, sep="\t", fill=T) %>% filter(V1 %in% genes$V1)
genes <- dim(genes)[1]
tibble(Type=c("Number of genes","Percentage of genes"),
       Annotated_genes=c(length(unique(pfam_annot$V1)),length(unique(pfam_annot$V1))/genes*100),
       Not_annotated_genes=c((genes - length(unique(pfam_annot$V1))),(genes - length(unique(pfam_annot$V1)))/genes*100)) %>% kable(digits=3, format.args = list(big.mark = ",")) %>%
  kable_styling(bootstrap_options = c("striped", "condensed"),full_width = F)
```

#### Detection of spurious and shadow genes

To detect eventual spurious genes we applied two different approaches. First we screened the data set against the AntiFam database [[4]](#4), which contains Pfam protein families "composed solely of spurious open reading frames (ORFs). Second, we identified the so-called "shadow ORFs", artifacts produced during the identification of the coding region that defines an ORF and consisting of intervals overlapping the coding region, following the criteria applied in Yooseph et al., 2018 [[5]](#5):
a) Two genes on the same strand are considered shadows if they overlap by at least 60 bps.
b) genes on opposite strands are identified as shadows if they overlap by at least 50 bps, and their ends of 3' are within the intervals of the others, or if they overlap by at least 120 bps and the end of 5' of one is in the interval of the other.

```{r,echo=FALSE}
sp_sh <- fread(paste0(dir,"/spurious_shadow/spurious_shadow_info.tsv"),stringsAsFactors = F,header = F)
spurious <- dim(sp_sh %>% filter(V7==TRUE))[1]
shadows <- dim(sp_sh %>% filter(V6==TRUE))[1]

tibble(Type=c("Number of genes","Percentage of genes"),
       Spurious_genes=c(spurious,spurious/genes*100),
       Shadow_genes=c(shadows,shadows/genes*100)) %>% kable(digits=3, format.args = list(big.mark = ",")) %>%
  kable_styling(bootstrap_options = c("striped", "condensed"),full_width = F)
```

We then concatenated the original and the new spurious and shadow genes information, needed by followin workflow steps.

### Clustering update

To integrate the new data in the original cluster DB, we performed an incremental clustering using the MMSeqs2 (Many-against-Many sequence searching 2) program _clusterupdate_ [[6]](#6), and applying the same parameters used for the original clustering.

The table below shows the results of the clustering update.

```{r,echo=FALSE,warning=FALSE}
# Add new clusters...divsion original - shared - new_data
new <- fread(paste0(dir,"/mmseqs_clustering/cluDB_new_name_rep_size.tsv"),stringsAsFactors = F,header = F,sep="\t")
shared = fread(paste0(dir,"/mmseqs_clustering/cluDB_shared_name_rep_size.tsv"),stringsAsFactors = F,header = F,sep="\t")
original = fread(paste0(dir,"/mmseqs_clustering/cluDB_original_name_rep_size.tsv"),stringsAsFactors = F,header = F,sep="\t")

tibble(Total_number_of_clusters=dim(new)[1] + dim(shared)[1] + dim(original)[1],
       Total_number_of_genes=sum(as.numeric(new$V3)) + sum(as.numeric(shared$V3)) + sum(as.numeric(original$V3)),
       Original = dim(original)[1],
       Shared = dim(shared)[1],
       New = dim(new)[1]) %>% kable(digits=3, format.args = list(big.mark = ",")) %>%
  kable_styling(bootstrap_options = c("striped", "condensed"),full_width = F)

```

The new clusters and the previous singletons that are now shared cluster with more than 1 gene, will be processed through the workflow. The new dataset that will be validated and classified is described in the following table:

```{r,echo=FALSE}
# Then the new set that is evaluated: cluster - singletons
clu <- fread(paste0(dir,"/mmseqs_clustering/cluDB_name_rep_size.tsv"),stringsAsFactors = F,header = F)

tibble(Total_number_of_clusters=dim(clu)[1],
       Singletons = dim(clu %>% filter(V3==1))[1],
       Clusters = dim(clu %>% filter(V3>1))[1],
       Cluster_genes = sum(clu %>% filter(V3>1) %>% .$V3)) %>% kable(digits=3, format.args = list(big.mark = ",")) %>%
  kable_styling(bootstrap_options = c("striped", "condensed"),full_width = F)

```

```{r,echo=FALSE,results='asis'}
avg <- mean(clu$V3)

if (round(avg,0)==1) {
  avg <- round(avg,0) + 1
}else{
  avg <- round(avg,0)
}

cat(paste("The mean cluster size is",avg,"genes"))
```

**Annotations in the clusters**

We concatebated the original and the new Pfam annotations.
We distributed the Pfam annotations in the clusters, to retrieve the position of the annotated sequences in the clusters.

```{r,echo=FALSE}
clu_annot <- fread(paste0(dir,"/annot_and_clust/annotated_clusters.tsv"),
                 stringsAsFactors = F,header = F)
cl_annot <- length(unique(clu_annot$V2))
orf_annot <- dim(clu_annot)[1]

tibble(Type=c("Clusters","Genes"),
       Annotated_clusters=c(cl_annot,orf_annot),
       Not_annotated_clusters=c(dim(clu %>% filter(V3>1))[1]-cl_annot,
       sum(clu %>% filter(V3>1)  %>% .$V3)-orf_annot)) %>% kable(digits=3, format.args = list(big.mark = ",")) %>%
  kable_styling(bootstrap_options = c("striped", "condensed"),full_width = F)

```

### Validation

**Compositional validation**

-   Evaluation of cluster homogeneity at the sequence level, based on an estimation of the homologous and non-homologous relations between sequences in a protein multiple alignment.

We performed a multiple sequence alignment (MSA) within each cluster to identify incongruous sequences, using the FAMSA fast multiple alignments program [[7]](#7) (version 1.1). We evaluated the MSAs of clusters using LEON-BIS [[8]](#8), a method that allows the accurate detection of conserved regions in multiple sequence alignments via Bayesian statistics, identifying the non-homologous sequences with respect to the cluster representative.

We considered "compositionally bad" those clusters with a percentage of non-homologous genes higher than the threshold detected using a broken-stick model.

```{r,echo=FALSE}
val <- fread(paste0(dir,"/validation/validation_results_stats.tsv",sep=""),
                 stringsAsFactors = F,header = T)

tibble(Type=c("Number of clusters","Percentage of clusters"),
       Compositional_good_clusters=c(val$comp_good,val$comp_good/val$total_clusters*100),
       Compositional_bad_clusters=c(val$comp_bad,val$comp_bad/val$total_clusters*100)) %>% kable(digits=3, format.args = list(big.mark = ",")) %>%
  kable_styling(bootstrap_options = c("striped", "condensed"),full_width = F)

```

```{r,echo=FALSE,warning=FALSE, fig.width = 9, fig.height=5}

load(paste0(dir,"/validation/compositional_validation_plots.rda"))

rej_thr <- p_rej_clp$data %>% spread(class,clusters) %>% filter(threshold < 0.9) %>%
 arrange(desc(threshold)) %>% filter(Observed>BSM) %>% slice(1)

tibble(Type=c("Non-homologous genes threshold","Number of clusters identified as bad"),
       rej_threshold=c(rej_thr$threshold,rej_thr$Observed)) %>% kable(digits=3, format.args = list(big.mark = ",")) %>%
  kable_styling(bootstrap_options = c("striped", "condensed"),full_width = F)

# p_rej_clp #Not working (Error in zero_range(to) : x must be length 1 or 2)
ggplot(p_rej_cl_bsm %>% as_tibble(), aes(threshold, clusters, fill=class)) +
  geom_col(position ="dodge", color = "black", alpha = 0.7) +
  geom_line(aes(group = class, color = class), position=position_dodge(width=0.9)) +
  geom_point(position=position_dodge(width=0.9), colour="black",  shape = 21) +
  ggrepel::geom_label_repel(data=(p_rej_cl_bsm %>% filter(threshold==rej_thr$threshold) %>%
    mutate(label=ifelse(class=="Observed",paste0("T=",threshold,"; N=",clusters),NA))),
                            aes(label = label), size = 4, position = position_dodge(width=0.9),
                            box.padding = unit(0.4, "lines"),point.padding = unit(0.45, "lines"),show.legend = F,fill="white") +
  theme_light() +
  xlab("Proportion of non-homolog ORFs per cluster") +
  ylab("Number of clusters") +
  scale_color_manual(values=c("#94bacc","#d95472")) +
  scale_fill_manual(values=c("#94bacc","#d95472")) +
  scale_y_continuous(labels = scales::comma) +
  scale_x_discrete(breaks = unique(p_rej_cl_bsm$threshold[seq(1, length(unique(p_rej_cl_bsm$threshold)), by = 10)])) +
  theme(axis.text = element_text(size=12),
        axis.title = element_text(size=13),
        legend.title = element_blank())

```

**Functional validation**

-   Investigation of the functional composition of the annotated clusters, applying a combination of text mining (Shingling) and Jaccard similarity.
We considered the different order of multi-domain annotation on the genes.

We considered "functionally bad" those clusters with an average Jaccard similarity value < 1.

```{r,echo=FALSE}
tibble(Type=c("Number of clusters","Percentage of clusters"),
       Functional_good_clusters=c(val$func_good,val$func_good/val$total_clusters*100),
       Functional_bad_clusters=c(val$func_bad,val$func_bad/val$total_clusters*100)) %>% kable(digits=3, format.args = list(big.mark = ",")) %>%
  kable_styling(bootstrap_options = c("striped", "condensed"),full_width = F)

```

**Validation combined results**

```{r,echo=FALSE}

tibble(Type=c("Number of clusters","Percentage of clusters","Number of genes","Percentage of genes"),
       Good=c(val$good_cl_n, val$good_cl_n/val$total_clusters*100, val$good_cl_orfs, val$good_cl_orfs/val$total_orfs*100),
       Bad=c(val$bad_cl_n, val$bad_cl_n/val$total_clusters*100, val$bad_cl_orfs, val$bad_cl_orfs/val$total_orfs*100)) %>% kable(digits=3, format.args = list(big.mark = ",")) %>%
  kable_styling(bootstrap_options = c("striped", "condensed"),full_width = F)

```

```{r,echo=FALSE,warning=FALSE, fig.width = 4, fig.height=4}
cl_desc <- data.frame(class = c("Bad clusters", "Good clusters"), num = c(val$bad_cl_n, val$good_cl_n))

ggplot(cl_desc, aes(class, num)) +
  geom_bar(stat = "identity") +
  scale_y_continuous(labels = scales::comma) +
  ylab("Number of clusters") +
  xlab("") +
  theme_light() +
  theme(plot.title = element_text(size=13),
        axis.text = element_text(size=11),
        axis.title = element_text(size=12))
```

### Refinement

After the validation, we proceeded with the retrieval of a subset of high-quality clusters. At first, we removed the clusters containing a number of non-homologous genes over the detected threshold (see above) and a Jaccard similarity index < 1, and the clusters with more than 30% shadow genes. From the remaining set of clusters, we removed the single shadow, spurious and/or non-homologous genes.


```{r,echo=FALSE}
ref <- fread(paste(dir,"/cluster_refinement/refined_clusters.tsv",sep=""),
                 stringsAsFactors = F,header = F)

tibble(Type=c("Number of clusters","Percentage of clusters","Number of genes","Percentage of genes"),
       Refined=c(length(unique(ref$V1)),length(unique(ref$V1))/val$good_cl_n*100,
       dim(ref)[1],dim(ref)[1]/val$good_cl_orfs*100),
       Discrded=c(val$good_cl_n-length(unique(ref$V1)),(val$good_cl_n-length(unique(ref$V1)))/val$good_cl_n*100,
                  val$good_cl_orfs-dim(ref)[1], (val$good_cl_orfs-dim(ref)[1])/val$good_cl_orfs*100)) %>% kable(digits=3, format.args = list(big.mark = ",")) %>%
  kable_styling(bootstrap_options = c("striped", "condensed"),full_width = F)

```

```{r,echo=FALSE,results='asis'}
ref_annot <- dim(fread(paste(dir,"/cluster_refinement/refined_annotated_clusters.tsv",sep=""),
                 stringsAsFactors = F,header = F) %>% select(V1) %>% distinct())[1]

ref_noannot <- dim(fread(paste(dir,"/cluster_refinement/refined_not_annotated_clusters.tsv",sep=""),
                 stringsAsFactors = F,header = F) %>% select(V1) %>% distinct())[1]

cat(paste("Refined annotated clusters: ", ref_annot,"\n\n"))

cat(paste("Refined not annotated clusters: ", ref_noannot))
```

```{r,echo=FALSE,warning=FALSE, fig.width = 4, fig.height=4}
cl_ref <- data.frame(class = c("Discarded", "Kept_refined"),
                     num = c(val$good_cl_n-length(unique(ref$V1)),length(unique(ref$V1))))

ggplot(cl_ref, aes(class, num)) +
  geom_bar(stat = "identity") +
  scale_y_continuous(labels = scales::comma) +
  ylab("Number of clusters") +
  xlab("") +
  theme_light() +
  theme(plot.title = element_text(size=13),
        axis.text = element_text(size=11),
        axis.title = element_text(size=12))
```

### Classification (cluster categories) & refinement

**Cluster categories:**

-   Knowns with PFAM (Ks): gene clusters annotated with PFAM domains of known function.
-   Knowns without PFAMs (KWPs): gene clusters that have a known function, but do not contain PFAM annotations.
-   Genomic unknowns (GUs): gene clusters that have an unknown function (e.g., DUF, hypothetical protein) but found in sequenced or draft-genomes.
-   Environmental unknowns (EUs): gene clusters of unknown function not found in sequenced or draft genomes, but only in environmental metagenomes.

**Clusters of knowns:**

We parsed the subset of annotated refined clusters to retrieve a single representative domain architecture for each cluster.

**Clusters of unknowns:**

1) *Identifying the GUs*: we searched the cluster consensus sequences against the Uniref90 database (release 2017_11) [[9]](#9), using MMseqs (e-value ≤ 1e-5, coverage ≥ 0.6). We classified a cluster as GUs if all the consensus sequence hits, falling within the 60% of the best log10 e-value, resulted annotated to a list of terms, such as "hypothetical" and "uncharacterized" , Pfam Domains of Unknown Function (DUF), etc., commonly used to define protein of unknown function in public databases. The non-annotated clusters with a match to characterised proteins were instead considered KWPs.
2) *Identifying the EUs*: we searched the consensus without a match to UniRef90 entries against the NCBI non-redundant database (release 2017_12) [[10]](#10), using MMseqs (e-value ≤ 1e-5, coverage ≥ 0.6). The queries without any match to nr entries, represent the EUs set. The consensus sequences matching nr "hypothetical" entries, i.e. those entries with descriptions corresponding the list of terms mentioned above, were considered GUs, while those annotated to characterised entries, KWPs.
3) *Unknown refinement*: we searched the EUs against the Uniclust database (release 30_2017_10), in search of remote homologies, using HHblits [[11]](#11). The results were parsed using a probability threshold of 90% and then processed with the same system used for the classification of the unknowns to retrieve the hits annotated to hypothetical or characterised proteins. The firsts were then moved to the GUs and the second to the KWPs. The clusters with no remote homologies represented the refined set of EUs.
4) *Known refinement*: we searched the KWPs profiles against the Pfam database (v.31) in search of remote homologies, using HHblits [[11]](#11). The results were parsed using a probability threshold of 90% and a target coverage of 60%. We removed the overlapping Pfam domains. We took into account the position of the Pfam domain / multi-domain annotations. Clusters annotated to DUFs were moved to the GUs, while those annotated to Pfam domain of known function to the Ks.

**Results**

```{r,echo=FALSE, warning=FALSE}
k1 <- dim(fread(paste(dir,"/cluster_classification/k_ids.txt",sep=""),
                 stringsAsFactors = F,header = F))[1]
kwp1 <- dim(fread(paste(dir,"/cluster_classification/kwp_ids.txt",sep=""),
                 stringsAsFactors = F,header = F))[1]
gu1 <- dim(fread(paste(dir,"/cluster_classification/gu_ids.txt",sep=""),
                 stringsAsFactors = F,header = F))[1]
eu1 <- dim(fread(paste(dir,"/cluster_classification/eu_ids.txt",sep=""),
                 stringsAsFactors = F,header = F))[1]
kwp2 <- dim(fread(paste(dir,"/cluster_categories/eu_hhbl_new_kwp_ids.txt",sep=""),
                 stringsAsFactors = F,header = F))[1]
gu2 <- dim(fread(paste(dir,"/cluster_categories/eu_hhbl_new_gu_ids.txt",sep=""),
                 stringsAsFactors = F,header = F))[1]
eu2 <- dim(fread(paste(dir,"/cluster_categories/eu_ids.txt",sep=""),
                 stringsAsFactors = F,header = F))[1]

kwp3 <- kwp1 + kwp2
gu3 <- gu1 + gu2
eu3 <- eu2
eu2 <- kwp2 + gu2

k2 <- length(unique(fread(paste0(dir,"/cluster_categories/kwp_hhbl_new_k_ids_annot.tsv"),
                 stringsAsFactors = F,header = F)$V1))
gu4 <- length(unique(fread(paste0(dir,"/cluster_categories/kwp_hhbl_new_gu_ids_annot.tsv"),
                 stringsAsFactors = F,header = F)$V1))

kwp4 <- kwp3 - (k2 + gu4)
gu5 <- gu3 + gu4
k3 <- k1 + k2

options(knitr.table.format = 'markdown')
tibble(Number_of_clusters=c("Pre-unknown_refinement","Unknown_refinement", "Post-unknown_refinement"),
       K=c(k1," ",paste(k1," (",k1/length(unique(ref$V1))*100,"%)")),
       KWP=c(kwp1,paste("+",kwp2),paste(kwp3," (",kwp3/length(unique(ref$V1))*100,"%)")),
       GU=c(gu1,paste("+",gu2),paste(gu3," (",gu3/length(unique(ref$V1))*100,"%)")),
       EU=c(eu1,paste("-",eu2),paste(eu3," (",eu3/length(unique(ref$V1))*100,"%)"))) %>%
       kable(digits=3, format.args = list(big.mark = ",")) %>%
  kable_styling(bootstrap_options = c("striped", "condensed"),full_width = F)

tibble(Number_of_clusters=c("Pre-known_refinement","Known_refinement", "Post-known_refinement"),
       K=c(k1,paste("+",k2),paste(k3," (",k3/length(unique(ref$V1))*100,"%)")),
       KWP=c(kwp3,paste("-",(k2+gu4)),paste(kwp4," (",kwp4/length(unique(ref$V1))*100,"%)")),
       GU=c(gu3,paste("+",gu4),paste(gu5," (",gu5/length(unique(ref$V1))*100,"%)")),
       EU=c(eu3," ",paste(eu3," (",eu3/length(unique(ref$V1))*100,"%)"))) %>%
       kable(digits=3, format.args = list(big.mark = ",")) %>%
  kable_styling(bootstrap_options = c("striped", "condensed"),full_width = F)
```

```{r,echo=FALSE,warning=FALSE, fig.width = 5, fig.height=3}
cl_categ <- data.frame(categ = c("K", "KWP","GU","EU"),
                       num = c(k3,kwp4,gu5,eu3),
                       perc = c(k3/length(unique(ref$V1)),kwp4/length(unique(ref$V1)),
                              gu5/length(unique(ref$V1)),eu3/length(unique(ref$V1))))

cl_categ$categ <- factor(cl_categ$categ, levels = c("EU", "GU", "KWP", "K"))
ggplot(cl_categ, aes(categ, perc, fill=categ)) +
  geom_bar(stat = "identity") +
  coord_flip() +
  scale_y_continuous(labels = scales::percent, breaks = seq(0,0.50,0.20)) +
  ylab("Clusters") +
  xlab("") +
  theme_light() +
  scale_fill_manual(values = c("#E84646","#65ADC2","#556c74", "#233B43")) +
  theme(plot.title = element_text(size=13),
        axis.text = element_text(size=11),
        axis.title = element_text(size=12),
        legend.position = NULL)
```

### Cluster communities

We aggregated our gene clusters into cluster communities to obtain larger groups, which take into account gene distant homologies.
The aggregation process comines remote homologies detection among the clusters of the different categories via HHBLITS [[11]](#11) searches, graph commmunity detection with the MCL algorithm [[12]](#12) and the evaluation of the cluster consensus domain architectures.

```{r,echo=FALSE,warning=FALSE}

k <- fread(cmd=paste0("cat ", dir, "/cluster_communities/*/k_partition_stats_eval_data_*.tsv"),
 stringsAsFactors = F, header = T,verbose=F)


best_inflation <- k %>%
  ungroup() %>%
  select(grp, total) %>%
  unique() %>%
  top_n(n = 1, wt = total) %>%
  .$grp

if (length(best_inflation >1 )) {
  best_inflation <- max(best_inflation)
}

cat(paste("The best inflation value was found at: ",best_inflation))

```

```{r,echo=FALSE,warning=FALSE}
comm <- fread(paste0(dir,"/cluster_communities/cluster_communities.tsv"),
                 stringsAsFactors = F,header = T)

tibble(Type=c("Communities","Clusters"),
       K = c(dim(comm %>% filter(category=="k") %>% select(com) %>% distinct())[1], dim(comm %>% filter(category=="k"))[1]),
       KWP = c(dim(comm %>% filter(category=="kwp") %>% select(com) %>% distinct())[1], dim(comm %>% filter(category=="kwp"))[1]),
       GU = c(dim(comm %>% filter(category=="gu") %>% select(com) %>% distinct())[1], dim(comm %>% filter(category=="gu"))[1]),
       EU = c(dim(comm %>% filter(category=="eu") %>% select(com) %>% distinct())[1], dim(comm %>% filter(category=="eu"))[1]),
       TOTAL = c(dim(comm %>% select(com) %>% distinct())[1], dim(comm)[1])) %>% kable(digits=3, format.args = list(big.mark = ",")) %>%
kable_styling(bootstrap_options = c("striped", "condensed"),full_width = F)

```

### Final integration of the original and new cluster DBs

We integrated the main results for the original and newly obtained cluster DBs. The main numbers are reported in the following tables.

```{r,echo=FALSE,warning=FALSE}
clu_comm <- fread(paste0(dir,"/integrated_cluster_DB/cluster_communities.tsv"),
                 stringsAsFactors = F,header = T)

tibble(Type=c("Communities","Clusters"),
       K = c(dim(clu_comm %>% filter(category=="k") %>% select(com) %>% distinct())[1], dim(clu_comm %>% filter(category=="k"))[1]),
       KWP = c(dim(clu_comm %>% filter(category=="kwp") %>% select(com) %>% distinct())[1], dim(clu_comm %>% filter(category=="kwp"))[1]),
       GU = c(dim(clu_comm %>% filter(category=="gu") %>% select(com) %>% distinct())[1], dim(clu_comm %>% filter(category=="gu"))[1]),
       EU = c(dim(clu_comm %>% filter(category=="eu") %>% select(com) %>% distinct())[1], dim(clu_comm %>% filter(category=="eu"))[1]),
       TOTAL = c(dim(clu_comm %>% select(com) %>% distinct())[1], dim(clu_comm)[1])) %>% kable(digits=3, format.args = list(big.mark = ",")) %>%
kable_styling(bootstrap_options = c("striped", "condensed"),full_width = F)


HQ_clu <- fread(paste(dir,"/integrated_cluster_DB/HQ_clusters.tsv",sep=""),
                 stringsAsFactors = F,header = T)

tibble(Type=c("HQ clusters"),
      K = dim(HQ_clu %>% filter(category=="K"))[1],
      KWP = dim(HQ_clu %>% filter(category=="KWP") )[1],
      GU = dim(HQ_clu %>% filter(category=="GU"))[1],
      EU = dim(HQ_clu %>% filter(category=="EU"))[1],
      TOTAL = dim(HQ_clu)[1]) %>% kable(digits=3, format.args = list(big.mark = ",")) %>%
kable_styling(bootstrap_options = c("striped", "condensed"),full_width = F)

```


#### References

<a name="1"></a>[1] D. Hyatt, G.-L. Chen, P. F. Locascio, M. L. Land, F. W. Larimer, and L. J. Hauser, "Prodigal: prokaryotic gene recognition and translation initiation site identification.," BMC bioinformatics, vol. 11, p. 119, Mar. 2010.

<a name="2"></a>[2] R. D. Finn et al., "The Pfam protein families database: towards a more sustainable future," Nucleic Acids Research, vol. 44, no. D1, Jan. 2016.

<a name="3"></a>[3] R. D. Finn, J. Clements, and S. R. Eddy, "HMMER web server: interactive sequence similarity searching.," Nucleic acids research, vol. 39, no. Web Server issue, pp. W29"W37, Jul. 2011.

<a name="4"></a>[4] R. Y. Eberhardt, D. H. Haft, M. Punta, M. Martin, C. O'Donovan, and A. Bateman, "AntiFam: a tool to help identify spurious genes in protein annotation.," Database: the journal of biological databases and curation, vol. 2012, p. bas003, Mar. 2012.

<a name="5"></a>[5] S. Yooseph, W. Li, and G. Sutton, "Gene identification and protein classification in microbial metagenomic sequence data via incremental clustering.," BMC bioinformatics, vol. 9, p. 182, Apr. 2008.

<a name="6"></a>[6] M. Steinegger and J. Söding, "MMseqs2 enables sensitive protein sequence searching for the analysis of massive data sets.," Nature biotechnology, vol. 35, no. 11, pp. 1026"1028, Nov. 2017.

<a name="7"></a>[7] S. Deorowicz, A. Debudaj-Grabysz, and A. Gudy", "FAMSA: Fast and accurate multiple sequence alignment of huge protein families.," Scientific reports, vol. 6, p. 33964, Sep. 2016.

<a name="8"></a>[8]	R. Vanhoutreve, A. Kress, B. Legrand, H. Gass, O. Poch, and J. D. Thompson, "LEON-BIS: multiple alignment evaluation of sequence neighbours using a Bayesian inference system.," BMC bioinformatics, vol. 17, no. 1, p. 271, Jul. 2016.

<a name="9"></a>[9] M. Magrane and UniProt Consortium, "UniProt Knowledgebase: a hub of integrated protein data.," Database: the journal of biological databases and curation, vol. 2011, p. bar009, Mar. 2011.

<a name="10"></a>[10] NCBI Resource Coordinators, "Database resources of the National Center for Biotechnology Information.," Nucleic acids research, vol. 42, no. Database issue, pp. D7-17, Jan. 2014.

<a name="11"></a>[11] M. Remmert, A. Biegert, A. Hauser, and J. Söding, "HHblits: lightning-fast iterative protein sequence searching by HMM-HMM alignment.," Nat Methods, Nov. 2011.

<a name="12"></a>[12] Dongen, Stijn van, and Cei Abreu-Goodger. 2012. "Using MCL to Extract Clusters from Networks." In Bacterial Molecular Networks: Methods and Protocols, edited by Jacques van Helden, Ariane Toussaint, and Denis Thieffry, 281–95. New York, NY: Springer New York.
